src/pages/resultsView/plots/PlotsTab.tsx

```diff
@@ -476,6 +476,9 @@ export default class PlotsTab extends React.Component<IPlotsTabProps, {}> {
             .length;
         const vertAxisDataSampleCount = this.vertAxisDataPromise.result!.data
             .length;
+        if (!(this.plotDataExistsForTwoAxes || this.waterfallPlotIsShown)) {
```

Comment:
how about return null and a comment.  returning components is confusing for the reader (and could leader to potential issues above)